### PR TITLE
[Identity] Add additional CAE support

### PR DESCRIPTION
**Pull request by pvaneck**
Continous access evaluation support (CAE) was already supported in many of the credentials through MSAL (primarily user credentials). This just enables it in several other credential types such as ClientAssertionCredential and many of the Async Credentials.

To communicate that SDK apps can handle claims challenges, the credential auth request needs to have the following claim:
`{"access_token":{"xms_cc":{"values":["cp1"]}}}`

Passing this claim even if a resource provider doesn't support CAE or a tenant isn't configured for CAE will have no effect. 

More Info:
- https://learn.microsoft.com/en-us/azure/active-directory/develop/app-resilience-continuous-access-evaluation
- https://learn.microsoft.com/en-us/azure/active-directory/conditional-access/concept-continuous-access-evaluation-workload
- https://learn.microsoft.com/en-us/azure/active-directory/develop/claims-challenge

To-Do: 
- [x] Add tests

Closes: #26484

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_internal/aad_client_base.py [hunk 1]
```diff
@@ -310,6 +342,17 @@ def _post(self, data: Dict, **kwargs: Any) -> HttpRequest:
         return HttpRequest("POST", url, data=data, headers={"Content-Type": "application/x-www-form-urlencoded"})
 
 
+def _merge_claims_challenge_and_capabilities(capabilities, claims_challenge):
+    # Represent capabilities as {"access_token": {"xms_cc": {"values": capabilities}}}
+    # and then merge/add it into incoming claims
+    if not capabilities:
+        return claims_challenge
+    claims_dict = json.loads(claims_challenge) if claims_challenge else {}
+    for key in ["access_token"]:
```
Comment (xiangyan99) [inline sdk/identity/azure-identity/azure/identity/_internal/aad_client_base.py:351]: Why?

Comment (pvaneck) [reply sdk/identity/azure-identity/azure/identity/_internal/aad_client_base.py:351]: This was borrowed from the msal [implementation](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/dev/msal/application.py#L49) for handling claims + capabilities. Seems like there is the potential for adding the xms_cc value for "id_token", hence the loop.

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**